nearcache_map.go

```diff
@@ -319,3 +440,88 @@ func (ncm *nearCacheMap) handleBatchInvalidationMsg(rth *inearcache.RepairingHan
 	})
 	return rth.HandleBatch(keys, sources, partitions, seqs)
 }
+
+func (ncm *nearCacheMap) populateResultFromNearCache(keys []interface{}, entries []types.Entry) ([]interface{}, error) {
+	// see: com.hazelcast.client.map.impl.nearcache.NearCachedClientMapProxy#populateResultFromNearCache
+	var missKeys []interface{}
+	var i int
+	for _, k := range keys {
+		nk, err := ncm.toNearCacheKey(k)
+		if err != nil {
+			return nil, err
+		}
+		cached, ok, err := ncm.getCachedValue(nk, true)
+		if err != nil {
+			return nil, err
+		}
+		if ok && cached != nil {
+			entries[i] = types.Entry{Key: k, Value: cached}
+			i++
+			continue
+		}
+		missKeys = append(missKeys, k)
+	}
+	return missKeys, nil
+}
+
+func (ncm *nearCacheMap) getNearCacheReservations(keys []interface{}, keyDatas []serialization.Data) (map[inearcache.DataString]keyReservation, error) {
+	res := make(map[inearcache.DataString]keyReservation, len(keys))
+	for i, k := range keys {
+		kd := keyDatas[i]
+		rid, err := ncm.nc.TryReserveForUpdate(k, kd, inearcache.UpdateSemanticReadUpdate)
+		if err != nil {
+			return nil, err
+		}
+		if rid != inearcache.RecordNotReserved {
+			res[inearcache.DataString(kd)] = keyReservation{Key: k, ID: rid}
+		}
+	}
+	return res, nil
+}
+
+func (ncm *nearCacheMap) releaseRemainingReservedKeys(rks map[inearcache.DataString]keyReservation) {
+	// see: com.hazelcast.client.map.impl.nearcache.NearCachedClientMapProxy#releaseRemainingReservedKeys
+	serialize := ncm.serializeKeys
+	for kds, rk := range rks {
+		var key interface{}
+		if serialize {
+			key = kds
+		} else {
+			key = rk.Key
+		}
+		ncm.nc.Invalidate(key)
+	}
+}
+
+func (ncm *nearCacheMap) populateResultFromRemote(pairs []proto.Pair, entries []types.Entry, reservations map[inearcache.DataString]keyReservation) (int, error) {
+	// see: com.hazelcast.client.map.impl.nearcache.NearCachedClientMapProxy#populateResultFromRemote
+	// assumes entries has max capacity.
+	serialize := ncm.serializeKeys
+	i := len(entries) - len(pairs)
+	for _, p := range pairs {
+		kd := p.Key.(serialization.Data)
+		k, err := ncm.ss.ToObject(kd)
+		if err != nil {
+			return 0, err
+		}
+		e := types.Entry{Key: k}
```

Comment:
Java client assigns the keys according to the `serializeKeys` flag. This may be missing that switch

Reply:
That's returned to the user, and should have the deserialized key.